src/Core/Domain/Product/Image/QueryResult/Shop/ShopProductImagesCollection.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Image\QueryResult\Shop;
+
+class ShopProductImagesCollection
+{
+    private $shopProductImages;
+
+    public function __construct(ShopProductImages ...$shopProductImages)
+    {
+        $this->shopProductImages = $shopProductImages;
+    }
+
+    public function getShopProductImagesByShopId(int $shopId): ShopProductImages
+    {
+        $shopProductImages = $this->shopProductImages;
+        $shopProductImagesFiltered = array_filter(
+            $shopProductImages,
+            static function (ShopProductImages $shopProductImages) use ($shopId): bool {
+                return $shopProductImages->getShopId() === $shopId;
+            }
+        );
+        $shopProductImages = reset($shopProductImagesFiltered);
+
+        if ($shopProductImages === false) {
+            throw new \Exception('No ShopProductImages for shop id ' . $shopId);
```

Comment:
We can probably use a more fine-tuned exception, either ShopNotFoundException or if we want a less specific one `InvalidArgumentException`

Reply:
Which one do you prefer?

Reply:
It's no longer topical since we remove ` public function getShopProductImagesByShopId(int $shopId): ShopProductImages`